stack_overflow/src/main/java/com/gabrielcireap/stackOverflow/service/VoteManagementService.java

```diff
@@ -0,0 +1,187 @@
+package com.gabrielcireap.stackOverflow.service;
+
+import com.gabrielcireap.stackOverflow.entity.Answer;
+import com.gabrielcireap.stackOverflow.entity.Question;
+import com.gabrielcireap.stackOverflow.entity.Vote;
+import com.gabrielcireap.stackOverflow.exception.*;
+import com.gabrielcireap.stackOverflow.repository.RepositoryFactory;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class VoteManagementService {
+
+    private final RepositoryFactory repositoryFactory;
+    private final UserManagementService userManagementService;
+    private final AnswerManagementService answerManagementService;
+    private final QuestionManagementService questionManagementService;
+
+    @Transactional
+    public Vote save(Vote vote){
+        return repositoryFactory.createVoteRepository().save(vote);
+    }
+
+    @Transactional
+    public void remove(int id){
+        Vote vote = repositoryFactory.createVoteRepository().findById(id).orElseThrow(VoteNotFoundException::new);
+        repositoryFactory.createVoteRepository().remove(vote);
+    }
+
+    @Transactional
+    public Vote findByAnswerId(int answerId, int userId) {
+        return repositoryFactory.createVoteRepository().findByAnswerId(answerId, userId).orElseThrow(VoteNotFoundException::new);
+    }
+
+    @Transactional
+    public Vote findByQuestionId(int questionId, int userId){
+        return repositoryFactory.createVoteRepository().findByQuestionId(questionId, userId).orElseThrow(VoteNotFoundException::new);
+    }
+
+    @Transactional
+    public void upvoteAnswer(int answerId){
+        userManagementService.checkIfUserIsLogged();
+        Answer answer = answerManagementService.findById(answerId);
+        if(answer.getUser().equals(userManagementService.getLoggedUser())){
+            throw new VoteYourOwnException();
+        }
+
+        try{
+            Vote vote = findByAnswerId(answerId, userManagementService.getLoggedUser().getId());
+            if(vote.getIs_upvote()){
+                throw new UpvoteDuplicateException();
+            } else {
+                vote.setIs_upvote(true);
+                save(vote);
+
+                answer.setVoteCount(answer.getVoteCount() + 2);
+                answerManagementService.save(answer);
+
+                answer.getUser().setScore(answer.getUser().getScore() + 12);
```

Comment:
Generally, these constants for the scores would be declared at the beginning of the class (like `static final int ANSWER_UPVOTE_SCORE_BONUS = 12;`).